source/dynamical_systems/src/IDynamicalSystem.cpp

```diff
@@ -76,34 +77,34 @@ JointState IDynamicalSystem<JointState>::evaluate(const JointState& state) const
   return this->compute_dynamics(state);
 }
 
-template<class S>
-void IDynamicalSystem<S>::assert_parameter_exists(const std::string& name) {
-  if (this->param_map_.find(name) == this->param_map_.cend()) {
-    throw exceptions::InvalidParameterException("Could not find a parameter named '" + name + "'.");
-  }
-}
-
 template<class S>
 void IDynamicalSystem<S>::assert_parameter_valid(const std::string& name, StateType state_type) {
-  this->assert_parameter_exists(name);
   if (this->param_map_.at(name)->get_type() != state_type) {
     throw exceptions::InvalidParameterException("Parameter '" + name + "'exists, but has unexpected type.");
   }
 }
 
+template void IDynamicalSystem<CartesianState>::assert_parameter_valid(const std::string&, StateType);
```

Comment:
Do you put those because you don't want someone to do a `IDynamicalSystem<Jacobian>` or whatever? Because the annoying part is that you need to duplicate this for each of the accepted types for each functions. Usually we do that when you have an expected behavior for a certain type and another for another type. But here, you do exactly the same behavior for both `CartesianState` and `JointState` which are the only two accepted types. In that case you can just define the template function in the header. Any other instantiations of `IDynamicalSystem` with other type than `JointState` or `CartesianState` would fail anyway. And if we want to add `DualQuaternionState` for example then it would still work.

Reply:
I guess because I wanted to keep the header lightweight and then forgot that putting it there actually reduces code :) Corrected now